### DISCUSSION TITLE
What's the latest PROJ version that pyproj 2.1.3 can support?

### DISCUSSION THREAD

**Discussion by espellman** [pyproj4/pyproj#1353] [Q&A] [answered]
Pyproj versions have a well-documented, required **minimum** version of PROJ, but this question is the opposite. For a given pyproj version, what is the **maximum** version of PROJ that can work with?

In particular, I know that pyproj version 2.1.3 requires a **minimum** PROJ version of 6.0.0, but my question is can it work atop PROJ version 9.3.0 (or 8.x, or 7.x)?

For the purpose of this question, the meaning of "work" can be in the range of "it runs, but it will pass on PROJ 6+ convention changes which might bite unwary callers."

Thanks!

**Comment by djhoese** [comment, marked as answer]
@snowman2 will need to give any historical context (and any past experiences with PROJ releases breaking pyproj compatibility), but usually releases of pyproj use the newest version of PROJ available at the time (and even bundle it internally in the released wheels). As with any software project I'm not sure there can be any promise of "we tested it with version X and expect it to work with the next version too". That said, I would hope that a particular pyproj version works with any version of PROJ of the same major version. So if pyproj 3.x works with pyproj 9 (minimum) then I would hope it works with any version of PROJ 9.x. I would not assume that it works with PROJ 10.0.

You mention pyproj 2.1.3 which is very old (2019) so I wouldn't expect much from that as far as future compatibility with PROJ. pyproj 3.x you have a little more hope for.

Is there a reason you can't use a newer pyproj if you're able to get a newer PROJ?

**Reply by espellman** [reply]
Thank you so much. Great answer and very reasonable.

> Is there a reason you can't use a newer pyproj if you're able to get a newer PROJ?

We indeed plan to upgrade to a more recent pyproj, but we're trying to do it incrementally. Our original plan was to go "bottom up" and increment our very old PROJ while holding pyproj constant.

**Comment by snowman2** [comment]
There is a good chance that it will compile, but if it doesn't than it isn't compatible. If you would like to see if an older version of pyproj is fully compatible with a newer version of PROJ, there is a test suite available with the older version you can run to see if everything works.
